File: 백준/Silver/11866. 요세푸스 문제 0/요세푸스 문제 0.py
import sys
from collections import deque

N, K = map(int, sys.stdin.readline().split())

q = deque()

for i in range(N):
    q.append(i+1)
print("<", end="")
while q:
    for i in range(K - 1):
        temp = q.popleft()
        q.append(temp)
    if len(q) == 1:
        print(q.popleft(), end=">")
    else:
        print(q.popleft(), end=", ")

File: 백준/Silver/11866. 요세푸스 문제 0/요세푸스 문제 0.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String[] sa = br.readLine().split(" ");

        int N = Integer.parseInt(sa[0]);
        int K = Integer.parseInt(sa[1]) - 1;

        Queue<Integer> q = new LinkedList<>();

        for(int i=1; i<=N; i++){
            q.add(i);
        }

        StringBuilder sb = new StringBuilder();
        sb.append("<");

        int count = 0;
        while(!q.isEmpty()){
            if(count == K){
                count = 0;
                sb.append(q.poll());
                if(q.isEmpty()){
                    sb.append(">");
                    break;
                }
                else sb.append(", ");
            }else{
                int temp = q.poll();
                q.add(temp);
                count++;
            }
        }

        System.out.println(sb);

    }

}
